demoapp/demo_app.py

```diff
@@ -58,224 +58,430 @@ def get_mgl_sections_file() -> pd.DataFrame:
+            st.write(cb.total_tokens, cb.prompt_tokens, cb.completion_tokens, cb.total_cost)     
+    return response
     
-            with col1:
-                st.subheader(f"Original Bill: #{bill_number}")
-                st.write(bill_title)
-                st.write(bill_content)
-
-            with col2:
-                st.subheader("Generated Text")
-                st.write(response)
-                st.write("###")
-                    
-            with col3:
-                st.subheader("Evaluation Metrics")
-                # rouge score addition
-                scorer = rouge_scorer.RougeScorer(['rouge1', 'rouge2', 'rougeL'], use_stemmer=True)
-                rouge_scores = scorer.score(bill_content, response)
-                st.write(f"ROUGE-1 Score: {rouge_scores['rouge1'].fmeasure:.2f}")
-                st.write(f"ROUGE-2 Score: {rouge_scores['rouge2'].fmeasure:.2f}")
-                st.write(f"ROUGE-L Score: {rouge_scores['rougeL'].fmeasure:.2f}")
+def generate_response(df: pd.DataFrame, bill_number: str, bill_title: str, bill_text: str, mgl_ref: str) -> str:
+    """
+    Determines document size and generates an appropriate response for the bill.
+    
+    Args:
+        df (pd.DataFrame): DataFrame containing bill information.
+        bill_number (str): Number of the bill.
+        bill_title (str): Title of the bill.
+        bill_text (str): Text content of the bill.
+    
+    Returns:
+        str: Generated response for the bill.
+    """
+    API_KEY = st.session_state["OPENAI_API_KEY"]
+    os.environ['OPENAI_API_KEY'] = API_KEY
+    llm = ChatOpenAI(openai_api_key=API_KEY, temperature=0, model='gpt-4-1106-preview', model_kwargs={'seed': 42})  
+    
+    mgl_names = get_chap_sec_names(df, bill_number)
+    
+    #Context size for 'gpt-4-1106-preview' is 128K; Need to break up the documents and use vector embenddings when the word length of the documents is large
+    #Chose 90K cut off for large docs because: 1 token = 0.75 words --> 90000 words = ~120K words + leaving  room for response
+    if bill_text is None or str(bill_text).strip().lower() =="nan" or bill_text == "":
+        response = f"Bill {bill_number} is not available for summarization"
+    
+    elif (len(mgl_ref.split()) + len (bill_text.split())) < 90000:
+        response = generate_response_small_documents(bill_title, bill_text, mgl_ref, mgl_names, llm)
+    else:
+        response = generate_response_large_documents(bill_title, bill_text, mgl_ref, mgl_names, llm)
+    return response
+
+def combine_bill_text_mgl_and_names(bill_title:str, bill_text:str, mgl_ref:str):
+    """ 
+    Combines bill_title, bill_text and MGL_ref in one document for evaluation matrics
+    
+    Args:
+    bill_title: str
+    bill_text: str
+    mgl_ref: str
+    
+    Returns:
+    Combined text: str
+    """
+    combined_text = str(bill_title) + str(bill_text) + str(mgl_ref)
```

Comment:
Why not show the actual templated prompt instead of this concatenation?

Reply:
yeah, good idea. Changed the code to use query/prompt for evaluation metrics